server/src/main/java/org/apache/druid/metadata/segment/cache/HeapMemorySegmentMetadataCache.java

```diff
@@ -539,6 +552,7 @@ private long syncWithMetadataStore()
     updateUsedSegmentPayloadsInCache(datasourceToSummary);
     retrieveAllPendingSegments(datasourceToSummary);
     updatePendingSegmentsInCache(datasourceToSummary, syncStartTime);
+    retrieveAllSegmentSchemas(datasourceToSummary);
```

Comment:
Should add something to the method-level javadoc of "stuff that happens every sync".

Reply:
Do you mean the javadoc of each of these methods should say whether they are invoked in every sync or only the first sync? Or some other additional info too?

Reply:
Ah, right, will do 👍🏻